llvm_mode/set-counter-BBid-llvm-pass.so.cc

```diff
@@ -0,0 +1,79 @@
+#define AFL_LLVM_PASS
+
+#include "../config.h"
+#include "../debug.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fstream>
+#include<iostream>
+#include <inttypes.h>
+
+#include "llvm/ADT/Statistic.h"
+#include "llvm/IR/IRBuilder.h"
+#include "llvm/IR/LegacyPassManager.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/Debug.h"
+#include "llvm/Transforms/IPO/PassManagerBuilder.h"
+
+using namespace llvm;
+
+namespace {
+
+  class CounterBBid : public ModulePass {
+
+    public:
+
+      static char ID;
+      CounterBBid() : ModulePass(ID) { }
+      bool runOnModule(Module &M) override;
+
+  };
+
+}
+char CounterBBid::ID = 0;
+
+bool CounterBBid::runOnModule(Module &M) {
+
+  LLVMContext &C = M.getContext();
+
+  IntegerType *Int64Ty = IntegerType::getInt64Ty(C);
+
+  /* Instrument all the things! */
+
+  u64 block_counter = 1;
+  
+  for (auto &F : M){
+    for (auto &BB : F) {
+
+        assert(block_counter < MAP_SIZE && "counter is too large");
+        std::string value = std::to_string(block_counter);
+        MDString * S = MDString::get(C, value.c_str());
```

Comment:
add some `assert(S)` and `assert(Meta)`. Did you check the assert() is triggered? assert() is a noop if NDEBUG is set. You shuold check if this is the case or not.

Reply:
It is triggered, but it doesn't have the best UI to show the message! I can check manually and use FATAL instead if that sounds better.